index.html

```diff
@@ -3,47 +3,434 @@
+  document.getElementById('sort-asc').addEventListener('click', function () {
+    // Disable the sort if 'asc' was previously selected, otherwise set the sort to 'asc'
+    if (options.sorted === 'asc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'asc';
+    }
+
+    // And now we need to re-run the function that populates the table. Note that the results will be
+    // different because we've change the options object.
+    updateTable();
+  });
+
+  document.getElementById('sort-desc').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.sorted === 'desc') {
+      options.sorted = false;
+    } else {
+      options.sorted = 'desc';
+    }
+
+    updateTable();
+  });
+
+  // This is where we'll need to create new event listeners for our filter and next/previous page buttons.
+  // They'll look very similar to the above, just listening to different elements, and perhaps performing
+  // different actions when clicked.
+
+  document.getElementById('filter-Mgt').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'Mgt') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'Mgt';
+    }
+
+    updateTable();
+  });
+
+
+  document.getElementById('filter-Sr').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'Sr') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'Sr';
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('filter-Drs').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'Drs') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'Drs';
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('filter-Admin').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.filtered === 'Admin') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'Admin';
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('page-next').addEventListener('click', function () {
+    // Disable the sort if 'desc' was previously selected, otherwise set the sort to 'desc'
+    if (options.page === 0) {
+      options.page = 1;
+    } else if (options.page === 1) {
+      options.page = 2;
+    } else if (options.page === 2) {
+      options.page = 3
+    } else if (options.page === 3) {
+      options.page = 4;
+    } else if (options.page === 4) {
+      options.page = 5;
+    } else if (options.page === 5) {
+      options.page = 6;
+    }
```

Comment:
It hurts my eyes! 🔥👀🔥 

What about `options.page = options.page + 1`? Or to be more concise, we can even write `options.page += 1`. Of course we'll need to check that that page actually exists, but (hint), we won't know how many pages we have until after we've filtered the data in the `updateTable()` function.

Bonus for next week, instead of having this implicit disable where the buttons just stop working if we're on the last page, let's actually disable them. Alex showed quickly how to set this in D3 with `d3.selectAll(...).attr('disabled', true/false)`.